File: LeetCode/Mathematics/LC-1071 GCD of Strings/LC1071.java
class LC1071 {
    public static String gcdOfStrings(String str1, String str2) {
        if(!(str1 + str2).equals(str2 + str1))
            return "";
            
        int gcd = findGcd(str1.length(), str2.length());

        return str1.substring(0, gcd);
    }

    public static int findGcd(int a, int b) {
        return b == 0 ? a : findGcd(b, a%b);
    }

    public static void main(String[] args){
        System.out.println(gcdOfStrings("ABCABC", "ABC"));
        System.out.println(gcdOfStrings("ABABAB", "ABAB"));
        System.out.println(gcdOfStrings("LEET", "CODE"));
    }
}

File: LeetCode/Mathematics/LC-1071 GCD of Strings/LC1071.cs
using System;

class LC1071
{
    public static string GcdOfStrings(string str1, string str2)
    {
        if ((str1 + str2) != (str2 + str1))
            return "";

        int gcd = FindGcd(str1.Length, str2.Length);

        return str1.Substring(0, gcd);
    }

    public static int FindGcd(int a, int b)
    {
        return b == 0 ? a : FindGcd(b, a % b);
    }

    public static void Main(string[] args)
    {
        Console.WriteLine(GcdOfStrings("ABCABC", "ABC")); // Output: "ABC"
        Console.WriteLine(GcdOfStrings("ABABAB", "ABAB")); // Output: "ABAB"
        Console.WriteLine(GcdOfStrings("LEET", "CODE"));   // Output: ""
    }
}
